lua/nvim_lsp/metals.lua

```diff
@@ -0,0 +1,79 @@
+local skeleton = require 'nvim_lsp/skeleton'
+local util = require 'nvim_lsp/util'
+local lsp = vim.lsp
+local server_name = "metals"
+local bin_name = "metals"
+
+local function make_installer()
+  local install_dir = util.path.join{util.base_install_dir, server_name}
+  local metals_bin = util.path.join{install_dir, bin_name}
+  local X = {}
+  function X.install()
+    local install_info = X.info()
+    if install_info.is_installed then
+      print(name, "is already installed")
+      return
+    end
+    if not (util.has_bins("curl")) then
+      error('Need "curl" to install this.')
+      return
+    end
+    if not (util.has_bins("java")) then
+      error('Need "JDK" to install this.')
+      return
+    end
+    local coursier_bin = install_dir .. "/coursier"
+    local download_cmd = string.format("curl -fLo %s --create-dirs https://git.io/coursier", coursier_bin)
+    local chmod_cmd = string.format("chmod +x %s", coursier_bin)
+    local install_cmd = string.format("%s bootstrap --java-opt -Xss4m --java-opt -Xms100m --java-opt -Dmetals.client=coc.nvim org.scalameta:metals_2.12:0.7.6 -r bintray:scalacenter/releases -r sonatype:snapshots -o %s -f", coursier_bin, metals_bin)
```

Comment:
I'd recommend replacing the `0.7.6` here with a variable that is either defaulting to the latest stable or pulling in `metals.serverVersion` from the settings object.

Reply:
I will use `latest.release` parameter provided by coursier

https://github.com/coursier/coursier/blob/510ca407c060d23519592be74d20f70433d68f0d/modules/coursier/shared/src/test/scala/coursier/ResolveTests.scala#L152-L160